kong/runloop/certificate.lua

```diff
@@ -238,12 +238,27 @@ local function find_certificate(sni)
     return nil, err
   end
 
-  for _, sni, err in mlcache.each_bulk_res(res) do
+  for _, sni_new, err in mlcache.each_bulk_res(res) do
```

Comment:
Why change the var `sni` to `sni_new`?

Reply:
Because `sni` will be overwrited here if we don't give it a new name. And when `mlcache.each_bulk_res(res)` fails, the `sni` will be `nil`. So we can't get the original `sni` in the block.